jfr-mappers/src/main/java/com/newrelic/jfr/tometric/GCHeapSummaryMapper.java

```diff
@@ -21,21 +22,26 @@
   public List<? extends Metric> apply(RecordedEvent ev) {
     var timestamp = ev.getStartTime().toEpochMilli();
     long heapUsed = ev.getLong("heapUsed");
+
+    var attr = new Attributes();
+    var list = new ArrayList<Gauge>();
+
     RecordedObject heapSpace = ev.getValue("heapSpace");
-    long committedSize = heapSpace.getLong("committedSize");
-    long reservedSize = heapSpace.getLong("reservedSize");
-
-    var attr =
-        new Attributes()
-            .put("when", ev.getString("when"))
-            .put("heapStart", heapSpace.getLong("start"))
-            .put("committedEnd", heapSpace.getLong("committedEnd"))
-            .put("reservedEnd", heapSpace.getLong("reservedEnd"));
-
-    return List.of(
-        new Gauge("jfr.GCHeapSummary.heapUsed", heapUsed, timestamp, attr),
-        new Gauge("jfr.GCHeapSummary.heapCommittedSize", committedSize, timestamp, attr),
-        new Gauge("jfr.GCHeapSummary.reservedSize", reservedSize, timestamp, attr));
+    if (heapSpace != null) {
+      long committedSize = heapSpace.getLong("committedSize");
+      long reservedSize = heapSpace.getLong("reservedSize");
+
+      if (ev.getString("when") != null) {
+        attr.put("when", ev.getString("when"));
+      }
+      attr.put("heapStart", heapSpace.getLong("start"));
+      attr.put("committedEnd", heapSpace.getLong("committedEnd"));
+      attr.put("reservedEnd", heapSpace.getLong("reservedEnd"));
+      list.add(new Gauge("jfr.GCHeapSummary.heapCommittedSize", committedSize, timestamp, attr));
+      list.add(new Gauge("jfr.GCHeapSummary.reservedSize", reservedSize, timestamp, attr));
```

Comment:
This case is interesting because `Gauge` accepts a primitive `long` rather than a nullable boxed primitive `Long`. What the correct value to set when the data isn't available? Probably makes most sense to omit the entry from the response `list`.

Reply:
The Javadocs for those all say they never return null, so I think it's ok to just check heapSpace

Reply:
Hm... I see that `RecordedObject#getValue()` says:
```
IllegalArgumentException if the field doesn't exist, or the field value can't be converted to the type {@code long} via a widening conversion
```
And also
```
A field might change or be removed in a future JDK release. A best practice for callers of this method is to validate the field before attempting access.
```
I think its beyond the scope of this PR, but we should consider adding a set of helper methods to be able to access these fields in a safe way.